File: CMR-EC/11-08-23/SlidingWindow.java
class SlidingWindow{
    public static void main(String[] args) {
        int[] arr = {1,6,4,8,-9,87,54,-9};
        int k = 3;
        int ans = 0;
        int i=0,j=k-1;

        while(j<arr.length){
            if(i==0){
                for(int k1=i;k1<=j;k1++){
                    ans+=arr[k1];
                }
                i++;
                j++;
            }else{
                ans = Math.max(ans,ans-arr[i-1]+arr[j]);
                i+=1;
                j+=1;
            }
        }
        System.out.println(ans);
    }
}

File: capgemini/batch-3/08-02-24/SlidingWindow.py
def getMaximumSubarraySum(arr,k):
    i=0
    j = k-1
    ans = float("-inf")
    ps = 0
    while j<len(arr):
        currSum = 0
        if i ==0:
            for k in range(i,j+1):
                currSum+=arr[k]
                ps = currSum
        else:
            currSum = ps-arr[i-1]+arr[j]
            ps = currSum
        i+=1
        j+=1
        ans = max(ans,currSum)
    return ans




def main():
    arr = [2,-3,1,2,-5,-6,7,8,12]
    k = 3
    ans = getMaximumSubarraySum(arr,k)
    print(ans)

main()